src/main/java/mekanism/common/tile/prefab/TileEntityStructuralMultiblock.java

```diff
@@ -60,88 +98,61 @@ public boolean hasStructure(Structure structure) {
-                    if (structure.isValid()) {
-                        if (activeMultiblock == null && structure.getController() != null && getMultiblockData(structure).isFormed()) {
-                            activeMultiblock = entry.getKey().getNameLower();
-                        }
-                    } else {
-                        iterator.remove();
-                    }
-                }
-            }
-            if (ticker >= 3 && structures.isEmpty()) {
-                invalidStructure.tick(this, true);
-                //If we managed to find any structures check which one is active
-                for (Map.Entry<MultiblockManager<?>, Structure> entry : structures.entrySet()) {
-                    Structure structure = entry.getValue();
-                    if (structure.getController() != null && getMultiblockData(structure).isFormed()) {
-                        activeMultiblock = entry.getKey().getNameLower();
-                        break;
-                    }
-                }
-            }
-            // this could potentially fail if this structural multiblock tracks multiple structures, but 99.99% of the time this will be accurate
-            if (!Objects.equals(activeMultiblock, clientActiveMultiblock)) {
-                clientActiveMultiblock = activeMultiblock;
-                sendUpdatePacket = true;
-            }
-        }
-        return sendUpdatePacket;
+        return null;
+    }
+
+    @Override
+    public void onAdded() {
+        super.onAdded();
+        //Ensure placing a structural multiblock tries to form the connected multiblock
+        invalidStructure.tick(this, true);
     }
 
     @Override
     public InteractionResult onActivate(Player player, InteractionHand hand, ItemStack stack) {
-        for (Map.Entry<MultiblockManager<?>, Structure> entry : structures.entrySet()) {
-            Structure structure = entry.getValue();
-            IMultiblock<?> master = structure.getController();
-            if (master != null) {
-                MultiblockData data = getMultiblockData(structure);
-                if (data.isFormed() && structuralGuiAccessAllowed(entry.getKey().getNameLower())) {
-                    // make sure this block is on the structure first
-                    if (data.getBounds().getRelativeLocation(getBlockPos()).isWall()) {
-                        return master.onActivate(player, hand, stack);
+        if (!structuralGuiAccessAllowed()) {
+            //If we don't have any structures that allow gui access, just short circuit and pass
+            return InteractionResult.PASS;
+        }
+        InteractionResult result = InteractionResult.PASS;
+        for (Structure structure : structures.values()) {
+            //If we already have an interaction that has been handled with one of our multiblocks just pass
+            // in 99% of cases we will not have a second iteration of the loop
+            if (result == InteractionResult.PASS) {
```

Comment:
This would be better as an early return/break, mostly just because it would be easier to read